vircadia/vircadia-web-sdk
# Assignment client WebRTC connections

- WebRTC connections to assignment clients: Individual WebRTC data channel connections to each assignment client, with WebRTC signaling routed via the domain server.
- Add the start of the following assignment client APIs: `AudioMixer`, `AvatarMixer`, and `MessageMixer`.

Works in conjunction with Vircadia PR: https://github.com/vircadia/vircadia/pull/1313

#### QA

With the domain server running on localhost:
- Enable anonymous connections and restart.

In the directory where the PR files are checked out:
` `npm install` - To set up the the dev & test environment.
- `npm run lint` - Should complete with no errors.
- `npm run build` - Should complete with no errors.
- `npm run test` - All  tests should pass. 
    Run the tests against the domain server in https://github.com/vircadia/vircadia/pull/1313, configured to allow anonymous connections.
    Check that the domain-server.exe process doesn't crash and restart during the tests.
    Known issue: "Jest did not exit one second after the test run has completed." See: https://github.com/vircadia/vircadia-web-sdk/issues/21
- `npm run devdoc` : Docs should be created in `./docs/dev/` directory.
- `npm run sdkdoc` : Docs should be created in `./docs/sdk/` directory.
- Set up and run the example Web app per the README.md in the `./example` directory. Test with both Chromium-based and Firefox browsers.
  Note: If anonymous connection isn't permitted, the app will show a "REFUSED" state.
  Known issue: State reported by `DomainServer` isn't "ERROR" if location used is invalid.
